pkg/ebpf/c/tracer-ebpf.c

```diff
@@ -391,91 +356,72 @@ static void update_conn_stats(
 }
 
 __attribute__((always_inline))
-static void update_tcp_stats(
-    struct sock* sk,
-    tracer_status_t* status,
-    metadata_mask_t family,
-    u32 retransmits,
-    u64 ts) {
-    conn_tuple_t t = {};
-    tcp_stats_t* val;
+static void update_tcp_stats(conn_tuple_t* t, tcp_stats_t stats) {
+    // query stats without the PID from the tuple
+    u32 pid = t->pid;
+    t->pid = 0;
+
+    // initialize-if-no-exist the connetion state, and load it
```

Comment:
Do you know why we're doing this?

Reply:
I don't have any context, but looking at the code I'd say this is safer from a concurrency standpoint. Otherwise you might have two threads creating a new `tcp_stats_t` object concurrently and not properly aggregating the retransmits.

```
tcp_stats_t* val = bpf_map_lookup_elem(&tcp_stats, t);
t->pid = pid;
    if (val == NULL) {
        return;
    }